src/block.rs

```diff
@@ -68,15 +87,25 @@ pub struct ProcessingEVMBlock {
     signed_block: Option<SignedBlock>,
     block_traces: Option<Vec<TransactionTrace>>,
     contract_rows: Option<Vec<ContractRow>>,
-    pub transactions: Vec<Transaction>,
+    pub decoded_rows: Vec<DecodedRow>,
+    pub transactions: Vec<TelosEVMTransaction>,
+    pub new_gas_price: Option<U256>,
+    pub new_revision: Option<u32>,
+    pub new_wallets: Vec<WalletEvents>,
 }
 
 #[derive(Clone)]
 pub struct TelosEVMBlock {
     pub header: Header,
     pub block_num: u32,
     pub block_hash: B256,
-    pub transactions: Vec<Transaction>,
+    pub transactions: Vec<TelosEVMTransaction>,
+
+    pub new_gas_price: Option<U256>,
+    pub new_revision: Option<u32>,
+    pub new_wallets: Vec<WalletEvents>,
+    pub account_rows: Vec<AccountRow>,
+    pub account_state_rows: Vec<AccountStateRow>,
 }
 
 pub fn decode_raw(raw: &[u8]) -> RawAction {
```

Comment:
Instead of decoding every type in separate function we can do it like:

```rust
pub fn decode<T: Packer + Default>(raw: &[u8]) -> T {
    let mut result = T::default();
    result.unpack(raw);
    result
}
```

and then use it as:
```rust
            let raw: RawAction = decode(&action.data());
```

or
```rust
            let withdraw_action: WithdrawAction = decode(&action.data());
```

This can remove lot of duplication from the code. Better option would be if antelope provides generic way to unpack things 🤷  

Also, this affects:

```rust
                    if r.code == Name::new_from_str("eosio.evm") {
                        if r.table == Name::new_from_str("config") {
                            self.decoded_rows.push(DecodedRow::Config(decode(&r.value)));
                        } else if r.table == Name::new_from_str("account") {
                            self.decoded_rows
                                .push(DecodedRow::Account(decode(&r.value)));
                        } else if r.table == Name::new_from_str("accountstate") {
                            self.decoded_rows
                                .push(DecodedRow::AccountState(decode(&r.value)));
                        }
                    }
```

Reply:
The antelope code is ours, we could probably add it to the `StructPacker` macro, a "from" type function that unpacks from bytes?

We can do that later, for now I have implemented this cleaner solution from above.